lib/Twig/Extension/Core.php

```diff
@@ -976,7 +976,12 @@ function twig_escape_filter(Twig_Environment $env, $string, $strategy = 'html',
                 // \uHHHH
                 $char = twig_convert_encoding($char, 'UTF-16BE', 'UTF-8');
 
-                return '\\u'.strtoupper(substr('0000'.bin2hex($char), -4));
+                return implode('', array_map(
```

Comment:
I do agree with @nicolas-grekas although I think `str_pad` is more efficient than `sprintf` in this. Or use `substr` tricks:

```php
$char = strtoupper(bin2hex($char));

if (4 >= strlen($char)) {
  return '\u' . substr('0000' . $char, -4);
}

return '\u' . substr('0000' . $char, 0, -4) . '\u' . substr($char, -4);
```